harness/determined/common/api/experiment.py

```diff
@@ -217,11 +217,11 @@ def make_test_experiment_config(config: Dict[str, Any]) -> Dict[str, Any]:
                 "max_length": {"batches": 1},
             },
             "hyperparameters": generate_random_hparam_values(config.get("hyperparameters", {})),
-            "resources": {**config_test.get("resources", {"slots_per_trial": 1})},
```

Comment:
Apologies for the slow response but how does that fail since just after I added this:

```config_test["resources"]["slots_per_trial"] = 1```

Wouldn't that ensure that `config["resources"` always exists?

Reply:
no, if config_test["resources"] is not a dict already this would fail. for example:

```
In [1]: config_test = {}

In [2]: config_test["resources"]["slots_per_trial"] = 1
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-2-586ebcd75583> in <module>
----> 1 config_test["resources"]["slots_per_trial"] = 1

KeyError: 'resources'
```

that may happen when the input experiment config doesn't have the `resources` section.

Reply:
duh, of course, thanks, let me put in a fix

Reply:
Caveat: I'm not convinced we should change the current behavior, but if we do...

`config_test["resources"]["slots_per_trial"] = 1` is exactly the line that would fail if `config_test["resources"]` is None.

But rather than doing any more fancy oneliners, which this PR demonstrates can be the source of bugs, let's just do it simple-style:

```python
# handle resources not present, or resources is None
if not config_test.get("resources"):
    config_test["resources"] = {}
config_test["resources"]["slots_per_trial"] = 1
```